File: ruby/10_Summation_of_Primes.rb
#!/usr/bin/ruby
#The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

#Find the sum of all the primes below two million.
require 'prime'
def sumOfPrimes(max)
  sum = 0
  Prime.each() do |prime|
    break if prime > max
    sum += prime
  end
  sum
end

puts sumOfPrimes(2000000)

File: python/10_Summation_of_Primes.py
#!/usr/bin/python
"""Find the sum of all the primes below two million.
"""

from math import sqrt
def primer(testNum):
	for pcheck in range(3,(int(sqrt(testNum))+1),2):
		if(testNum%pcheck == 0):
			return False
	return True	


if __name__ == "__main__":

	sum = 2
	for index in range(3,2000000,2):
		if primer(index):
			sum += index
	print sum
